test_aodndata/gsla/test_gsla.py

```diff
@@ -0,0 +1,169 @@
+import os
+import unittest
+
+from aodncore.testlib import HandlerTestCase
+from aodncore.pipeline import PipelineFilePublishType, FileType, PipelineFile, PipelineFileCollection
+from aodncore.pipeline.storage import get_storage_broker
+from aodncore.pipeline.exceptions import InvalidFileFormatError, InvalidPathFunctionError, InvalidFileNameError
+from aodndata.gsla.handler import GslaHandler, GSLA_PREFIX_PATH, GSLA_REGEX, \
+    NC_CREATION_DATE, get_fields_from_filename, get_creation_date
+
+from datetime import datetime
+from mock import patch
+
+TEST_ROOT = os.path.join(os.path.dirname(__file__))
+
+
+GOOD_NC_DM00 = os.path.join(TEST_ROOT, "IMOS_OceanCurrent_HV_19930101T000000Z_GSLA_FV02_DM00_C-20130913T082343Z.nc.gz")
+GOOD_NC_NRT00 = os.path.join(TEST_ROOT, "IMOS_OceanCurrent_HV_20180101T000000Z_GSLA_FV02_NRT00_C-20180105T222006Z.nc.gz")
+GOOD_YEARLY_FILE = os.path.join(TEST_ROOT, "IMOS_OceanCurrent_HV_1993_C-20150521T030649Z.nc.gz")
```

Comment:
These files aren't included in the PR.

Reply:
Ok, I'm a bit confused to when file is required or not. The BAD_NC file for example is just about checking the file name so we don't really need to read the content of the file neither...

Reply:
It kind of depends really.
When the file doesn't exist physically in the test, it is because I only call the dest_path function from the handler class.

But if i want to run the Handler on the file, we NEED the file, as many checks will be done such as, is it a file, of which type ...

Reply:
Ok but do we need to run the handler when all we want to check is the file name? I'm just a bit reluctant in adding data files to the repo when we don't actually look at the data.

Reply:
we don't only want to check the file name. 
if you look at 
test_setup_upload_location_push_newer_file
test_setup_upload_location_push_older_file
for example,
A lot is being tested, and having the physical file is essential

Reply:
Yes I understand. But my point is all is done with bad.nc.gz is to check its file name. Why does it need to be in the repo?

Reply:
the size of bad.nc.gz is 0 byte. It is just created with ```touch bad.nc.gz``` so the handler can actually run on an existing file